src/network/networkd-dhcp4.c

```diff
@@ -579,7 +579,7 @@ int dhcp4_configure(Link *link) {
         assert(link->network->dhcp & ADDRESS_FAMILY_IPV4);
 
         if (!link->dhcp_client) {
-                r = sd_dhcp_client_new(&link->dhcp_client);
+                r = sd_dhcp_client_new(&link->dhcp_client, (int)link->network->dhcp_anonymize);
```

Comment:
the cast is redundant, the C compiler does that silently and automatically anyway